Initial softphone implementation for Chrome

jvanveen opened the pull request:
### Description of fix
Initial softphone implementation for Chrome using SIP.js, Vue & Vue-stash. Firefox and Electron builds will be fixed in sprints.

### Other info
Project doc: https://docs.google.com/document/d/1FxRPnTgT5Swwd2yZ-t64vN-9n19K0ITbWxgSGYCQUg0
Mockups: https://projects.invisionapp.com/share/T7F6YLXHX#/screens/

Commit:
Added Webrtc caller. Added Vue and Stash state management. Refactored event emitters. Refactored localstorage persistance.

Commit:
Added dialpad component.

Commit:
Add telemetry component. Refactoring state.

Commit:
Refactoring availability.

Commit:
Adding Vue to bg/fg app class.

Commit:
Refactoring scss.

Commit:
Adding Fields.

Commit:
Fixes settings page, adjust store for c2d.

Commit:
Added ringtone, bulmaified forms.

Commit:
Add package-lock.json

Commit:
Add calldialog.

Commit:
Added vm to bg script.

Commit:
Basic callstate flow using sip store.

Commit:
Keypad in callstate, fixed sourcemaps scss build.

Commit:
Flow fixes callstate dialog.

Commit:
Abstracting session.

Commit:
Cleanup sounds.

Commit:
Add initial ui design.

Commit:
Update package.json and package-lock.json.

Commit:
Hide keypad on finish call

Commit:
Add explicit stun server list.

Commit:
Fix display name and number shown in incoming dialog

Commit:
Queues and contacts fixups.

Commit:
Remove old code, so it doesnt get in the way during refactoring with multiple devs.

Commit:
Remove callstatus.

Commit:
Cleanup dead scss.

Commit:
Remove old widget state.

Commit:
Cleanup scss.

Commit:
Renamed navigation component to statusbar. Removed more legacy.

Commit:
More scss cleanup.

Commit:
Moved main entrypoint to app.

Commit:
Renaming build steps.

Commit:
Add statusbar call activity.

Commit:
Fix on-hold state and ui.

Commit:
Blind transfer.

Commit:
Fix presence.

Commit:
Remove logs.

Commit:
Fix for initial plugin build.

Commit:
osx height fix.

Commit:
Fix for chrome height issue on OSX

Commit:
Styling fixes.

Commit:
Fix incoming audio remote stream.

Commit:
Fixup.

Commit:
dependency update.

FrenchVillain committed:
Hide video elements

Commit:
Container fixes.

Commit:
Audio sinks, styling, etc.

Commit:
Status bar indicators.

Commit:
Update styling consistency.

Commit:
Translations and several small fixes.

Commit:
Disconnect sip on refresh.

Commit:
Fix vendor-specific box-shadow.

Commit:
Version bump.

Commit:
Propagate resetted state back to fg.

Commit:
Version bump.

Commit:
Refactor session class. Fix dtmf event.

Commit:
Add notification component to project.

Commit:
Cleanup components.

Commit:
Refactoring call states.

Commit:
Refactoring call and state handling from calls.

Commit:
Call state tracking using arrays.

Commit:
Fix basic actions on state.

Commit:
Cleanup setState.

Commit:
mergeDeep setState from fg using path.

Commit:
Basic UI for transfer.

Commit:
Basic call switching.

Commit:
Basic attended transfer flow.

Commit:
Extra wiring.

Commit:
Find contacts while typing.

Commit:
Small queues styling.

Commit:
Fixing height issues and visibility conditions.

Commit:
Start ringing on 180. Height/styling fixes.

Commit:
Call setState fix partial merge.

Commit:
Sanitize input, add transfer to group, dtmf fixes.

Commit:
Last minute fixes.

Commit:
Version bump.

Commit:
G722 only codec for calling

Commit:
Refactored sound. Added european ringback tone. Restyled navigation and notifications.

Commit:
Fix keyboard enter issue keypad.

Commit:
DTMF sound fix.

Commit:
Styling cleanup.

Commit:
Fix positioning notifications.

Commit:
Refactor module structure. Refactor contacts module.

Commit:
Keep track of popup open/closed status using port. Fix queue size timer. Contacts loading.

Commit:
Fix setting/unsetting active queue.

Commit:
Fix availability module. Clear state on db schema change.

Commit:
Generic styling fixes, queue size color indicator.

Commit:
Translations, styling, platform url helper, help texts.

Commit:
Fix resetting notification state. Fix availability edge case.

Commit:
Merge queues update in timer instead of resetting. Cleanup console logs. Version bump.

Commit:
Timers for queues fixed

Commit:
Set icon when logout is done

Commit:
Initial c2d code return.

Commit:
AVPF and encryption required checks

Commit:
Refactoring new calls paradigm, keypad.

Commit:
Adding calls new style.

Commit:
Basic click2dial.

Commit:
Added option to remove new call, styling flow.

Commit:
Transfer hinting, make sure at least one empty call is around when dialing with the dialpad component.

Commit:
Fix incoming calls.

Commit:
Push incoming call state.

Commit:
Vendor settings string, bulma-switch, improved availability setting logics, do not disturb switch.

Commit:
SCSS and Javascript cleanup. Naming convention fixes.

Commit:
Dnd statusbar.

Commit:
Red color

Commit:
Settings devided into tabs. Developer mode and developer-mode-only options.

Commit:
Add language selection.

Commit:
Added audio settings.

Commit:
Added bg watcher. Added menubar icon watcher.

Commit:
Cleanup menubar code. Added calling animation.

Commit:
Hint pulse, add busy tone.

Commit:
Friendlier translation for declined.

Commit:
Add a notification when a call ended.

Commit:
Call component styling/states cleanup.

Commit:
General polish to callstate and flow. Fixed blind transfer.

Commit:
Fix calling menubar event.

Commit:
Cleanup env. Only show notification when ui is not visible.

Commit:
Proper notification when call ends.

Commit:
Cleanup tabs=>observer handler.

Commit:
ConnectAB type call, making Call data structure more generic, added Vialer brand logos.

Commit:
Connectab error handling.

Commit:
Version bump.

Commit:
Keyboard shortcuts accept/decline call. Fixed incoming call regression.

Commit:
Update manifest. Remove description from browser action.

Commit:
Fix activeCall check. Version bump.

Commit:
Refactored utils/helpers, keyboard shortcuts and incoming/outgoing calls fixed again.

Commit:
Add dnd switch.

Commit:
Fix invalid hold check.

Commit:
Add extension module. State cleanup and watcher fix. Microphone permission fixes, ...

Commit:
No microphone access status in statusbar icon.

Commit:
feat: Disable keypad when Calling conditions are not met.

Commit:
feat: Added Telemetry component and events.

Commit:
feat: onboarding; select voipaccount from available accounts, set installed flag, reorder settings.

Commit:
fix: telemetry text changes, keypad disable condition fixes, voipaccount select visibility condition fix.

Commit:
fix: Cleanup helpers, cleanup component store mounts, added app store property, added vendor-specific builds. Added app name. Initial work to style the login page.

Commit:
feat: Add vendor support email option. More styling login component. Added svg logo and modified build.

Commit:
fix: adding svg icons, buildstep and updated styling.

Commit:
feat: replace all font icons with svg component. modified styling to resemble mockup new style.

Commit:
fix: cleanup redundant font stuff, streamline svg build with a temp dir. Split svg icons in generic and brand-specific.

Commit:
* Added: Notification box missing microphone permission
* Added: Notification box availability
* Added: Portal user settings link from Availability Notification box
* Added: Reject status distinction between reject_a and reject_b, e.g. show Call message about who actually ended the call
* Added: Show Call hang-up icon in callswitcher when call is being ended.
* Fixed: Reset transfer accept state on other calls when transfer source call is being ended by the transfer source callee.
* Changed: Allow to hangup a new call with status 'created' (not yet calling out)

Commit:
feat: add websocket retry delay, add presence throttle, moved app state to separate module, add online/offline state.

Commit:
fix: contacts unavailable styling, keyboard focus keypad, menubar icon height, forget pw link.

Commit:
Update gitignore.

Commit:
feat: settings and login field validation.

Commit:
feat: add alpha build target. make buildname customizable. remove invalid stunserver config.

Commit:
Alpha version fixes for Firefox build.

Commit:
fix: use correct camera disabled icon.

Commit:
fix: Obey click2dial setting.

Commit:
feat: removed jquery from vendor. replaced all selectors with querySelector. Fix calling through contextmenu.

Commit:
fix: cleanup observer, replace remaining $.each with iterator.

Commit:
fix: remove remaining jquery isArray check.

Commit:
fix: remove redundant tab script. Add disabled styling.

Commit:
fix: remove remaining logging.

Commit:
fix: adding logic to deal with disabled icons from the number of calls active.

Commit:
fix: cleanup notification after timeout.

Commit:
feat: unified notification for click-to-dial label notifications.

Commit:
fix: add same logic for tel: links.

Commit:
Move system notification decision to notification method.

Commit:
fix: unify call translations and messages.

Commit:
fix: cleanup label styling.

Commit:
feat: replace open popup with hold shortcut(ctrl-shift-x).

Commit:
version bump.

Commit:
fix: update docs & deps.

Commit:
fix: use component tag as custom outer element for components.

Commit:
fix: make component element display:block.

Commit:
fix: changing availability switch now propagates placeholder changes properly.
fix: propagating username/pw when switching softphone enable.
feat: notification about db schema change, and auto-logout after 5 seconds.
feat: notification-box text change when no availability options are present.
feat: notification-box availability about using refresh.
feat: change default ringtone.

Commit:
feat: deal with empty VoIP-account select. Added empty field option. Added notification box for empty situation.

Commit:
fix: always update token before performing autologin.

Commit:
fix: sentence with portal variable now has `the` before it.

Commit:
Version bump.

PM5544 reviewed (commented):
Don't really know what to add here since it involves so much code. Any specific thing to look for?

Commit:
feat: codec selection and functionality to make use of the new Opus feature in the VG backend.
feat: credential encryption using a PBKDF2 vault with optional auto-unlock.
feat: store persistance and revival with both encrypted and unencrypted parts.
feat: lock screen component with wiring to manually unlock the vault.
feat: add encryption key storage option in settings and info block to auto-unlock vault.
feat: use requestAnimationFrame for microphone meter and slower diff for more fluid response
feat: add privacy settings tab.
feat: extra slot for select field to add extra html (slightly modified styling ringtone field)
feat: add menubar icon that indicates unlocking is needed.
feat: show all contacts and subscribe only to those with sipreginfo.
feat: add unregistered status to contacts with additional styling.
feat: add css grid.
fix: allow subsequent call after ongoing call is in closing state, removing old call first.
fix: add crypto class to deal with symmetric and assymetric encryption.
fix: cleanup state handling code; only bg handles persistance now.
fix: login failure message after being blocked out of the platform by too many login attempts, instead of credentials message.
fix: remove redundant html pages
fix: make schema check and removal of old store data consistent.
fix: refactored field component
fix: cleanup webrtc account selection data-model
fix: softphone toggle cleans up selected voipaccount (deals with reconnection edge-cases)
fix: webrtc account initial value selection and validation conditions
fix: normalized telemetry permission text. telemetry now only shows after login.
fix: moved server url fields to general.
fix: moved click-to-dial icons field to calling(Renamed `phone` tab to `calling`)
fix: several translation improvements
fix: don't show validated field styling
fix: use green user icon for connectab instead of orange cloud icon
fix: useragent string formatting spacing and windows env detection on surfacebook (Win32)
fix: only show contact call & transfer icons for available contacts

Commit:
Merge branch 'feature/post-mvp-cleanup' into develop
